deepfence_utils/threatintel/common.go

```diff
@@ -123,19 +124,26 @@ func DeleteFileMinio(ctx context.Context, fName string) error {
 	return nil
 }
 
-func ExposeFile(ctx context.Context, fName string) (string, error) {
+func ExposeFile(ctx context.Context, fName string, consoleURL string, ttlCache *ttlcache.Cache[string, string]) (string, error) {
 
 	mc, err := directory.FileServerClient(directory.WithDatabaseContext(ctx))
 	if err != nil {
 		return "", err
 	}
 
-	url, err := mc.ExposeFile(ctx, fName, false, threatintelPollDuration*3, url.Values{})
-	if err != nil {
-		return "", err
+	var exposedURL string
+	cacheVal := ttlCache.Get(consoleURL + fName)
+	if cacheVal == nil {
+		exposedURL, err = mc.ExposeFile(ctx, fName, false, threatintelPollDuration*3, url.Values{}, consoleURL)
+		if err != nil {
+			return "", err
+		}
+		ttlCache.Set(consoleURL+fName, exposedURL, threatintelPollDuration)
```

Comment:
This operation is not thread safe - we might have multiple requests accessing the same thing

Reply:
Actually I confused `set` operation that is not safe with `Set` operation that is safe:
https://github.com/jellydator/ttlcache/blob/v3/cache.go#L313